src/validation.js

```diff
@@ -2,11 +2,18 @@ import _ from 'lodash/fp'
 import * as F from 'futil-js'
 import { mapAsync } from './util/promise'
 
-export let defaultHasValue = x => !F.isBlankDeep(_.some)(x.data)
+export let defaultHasValue = _.flow(
+  F.cascade(['data', 'values', 'value', 'query']),
```

Comment:
I don't think this is the right approach here, because it becomes so arbitrary. For example numbers have min/max - the core shouldn't have special knowledge about any of these properties. E.g. what happens if facet called it's optionsFilter `query` instead? Then by default, it'd assume it had a value leading to confusing bugs. The only reason we had data here is because we still support data as a magic property along with config - but perhaps the right answer is to just remove it entirely.

Reply:
If we remove data entirely we need to rework some unit tests, look at the feedback in this issue: https://github.com/smartprocure/contexture-client/issues/32 , essentially:

> If we remove it we'll need to remove unit tests like: https://github.com/smartprocure/contexture-client/blob/master/test/index.js#L112-L154

Reply:
yup, definitely need to rework tests :)

Reply:
The point is that types without a bucket where they throw in all there filtering "data" properties need to have manual hasValue functions

Reply:
gotcha

Reply:
@daedalus28 let's do something instead, we have a set of defaultTypes, let's have a defaultHasValue that has a list of the possible keys that can be checked for those default types. This would be immediately useful for us (since we won't need to define a custom types for those types, because we only use those types), but also immediately valuable for anyone starting to use this project! :) what do you think?

Reply:
per our Slack conversation, we'll have hasValue in each one of our example types and remove defaultHasValue